Adds guidelines for contributing and generating documentation (#390)
nteract/papermill: CONTRIBUTING.md

### Running Tests Locally

> If you are contributing with documentation please jump to the [next section.](#Running-Documentation-Tests)

Reviewer: I'd switch the wording up here:

```suggestion
If you are contributing with documentation please jump to [building documentation.](#Building-Documentation)
```

Now there should be a working and editable installation of Papermill to start making your own contributions.

> ### Running Documentation Tests

Reviewer: ```suggestion
### Building Documentation
```

> The documentation is built using the [Sphinx](http://www.sphinx-doc.org/en/master/) engine. To contribute, edit the [RestructuredText (`.rst`)](https://en.wikipedia.org/wiki/ReStructuredText) files to make changes and additions.

Reviewer 2: ```suggestion
The documentation is built using the [Sphinx](http://www.sphinx-doc.org/en/master/) engine. To contribute, edit the [RestructuredText (`.rst`)](https://en.wikipedia.org/wiki/ReStructuredText) files in the docs directory to make changes and additions.
```

> The documentation is build using the [Sphinx](http://www.sphinx-doc.org/en/master/) engine. To contribute, you can edit the `.rst` files (learn more about restructured text [here](https://en.wikipedia.org/wiki/ReStructuredText)).

Reviewer: ```suggestion
The documentation is built using the [Sphinx](http://www.sphinx-doc.org/en/master/) engine. To contribute, edit the [RestructuredText (`.rst`)](https://en.wikipedia.org/wiki/ReStructuredText) files to make changes and additions.
```

> Once you are done editing, to generate the documentation, just run the documentation specific tests.

Reviewer 2: ```suggestion
Once you are done editing, to generate the documentation, use tox and the following command from the root directory of the repository:
```
